### ISSUE TITLE
Force fullscreen

### ISSUE THREAD

**Issue by NOVstudios** [media-kit/media-kit#624]
Hello,

I haven't found a way to force the player to full screen at launch, does this exist?

I'd like to use the player only in full screen mode under certain conditions.
I've tried using _toggleFullscreen(context)_ to achieve this but I get an error _No [VideoStateInheritedWidget] found in [context]_.

More generally, what's the best way to switch to full screen from code?

**Comment by zezo357** [comment]
check https://github.com/zezo357/media_kit/blob/examples/media_kit_test/lib/tests/15.full_screen_player.dart

**Comment by NOVstudios** [comment]
Thank you very much, it works well. I've added media_kit/media_kit_test/lib/tests/ to my bookmarks! 

However, I have made a small adjustment.

I replaced :

```
if (key.currentState?.isFullscreen() ?? false) {
	key.currentState?.exitFullscreen();
}
Navigator.of(context).pop();
```

With :

```
if (key.currentState?.isFullscreen() ?? true) {
  key.currentState?.exitFullscreen().then((value) => Navigator.of(context).pop());
}else{
  Navigator.of(context).pop();
}
```


Because why exit fullscreen if you're not in fullscreen?
What's more, you have to wait until you're out of full screen before starting navigation to the previous page.

**Comment by zezo357** [comment]
here is an explained code to why `?? true` is wrong

### Original Function:
```dart
if (key.currentState?.isFullscreen() ?? false) { 
    // This block will not be executed if 'currentState' is null
    // or if 'isFullscreen()' returns false.
    key.currentState?.exitFullscreen(); 
}
```
In this function:
- The `if` condition checks if `currentState` is not `null` and if `isFullscreen()` returns `true`.
- If `currentState` is `null`, the expression after `??` (`false`) becomes the result, causing the block inside the `if` statement not to execute.
- If `currentState` is not `null` but `isFullscreen()` returns `false`, the block is also not executed.

### Modified Function:
```dart
if (key.currentState?.isFullscreen() ?? true) {
    // This block will be executed if 'currentState' is null 
    // or if 'isFullscreen()' returns true.
    key.currentState?.exitFullscreen().then((value) => Navigator.of(context).pop());
}
```
In this modified function:
- The `if` condition checks if `currentState` is `null` or if `isFullscreen()` returns `true`.
- If `currentState` is `null`, the expression after `??` (`true`) becomes the result, causing the block inside the `if` statement to execute.
- If `currentState` is not `null` and `isFullscreen()` returns `true`, the block is executed.
- The use of `?.` (null-aware method call) ensures that `exitFullscreen()` is only called if `currentState` is not `null`.
- The `then` part with `Navigator.of(context).pop()` will execute after `exitFullscreen()` completes, but only if `currentState` is not `null`.

In summary, the key difference between the two functions is how they handle the scenario when `currentState` is `null`. The original function does not execute the block when `currentState` is `null`, whereas the modified function does.



for the .then i think thats a good idea,

**Comment by NOVstudios** [comment]
Ok, thanks for the explanation, I'm still not familiar with the ?? notation.

**Comment by zezo357** [comment]
> Ok, thanks for the explanation, I'm still not familiar with the ?? notation.

It's okay this is why I explained it in full detail to avoid any confusion or misunderstanding.
